### PR TITLE
♻️ 1.10.0

### PR DESCRIPTION
**Pull request by EverlastingBugstopper**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,78 @@
 # Changelog
 
+## 1.10.0
+
+- ### Features
+
+  - **`wrangler dev` now requires that you specify "preview" versions of KV namespaces - [EverlastingBugstopper], [issue/1032] [pull/1357] [pull/1360]**
+
+    We're preparing to make `wrangler dev` run entirely on Cloudflare's edge, giving it access to functionality that's currently missing in the preview environment, and more accurately reproducing what happens when your Worker is deployed to production. This means that, among other things, your Workers can access the Cache API and your production KV Namespaces.
```
Comment (EverlastingBugstopper) [inline CHANGELOG.md:9]: i dont think this block is necessary. this affects users who arent running on the edge as well, as you can access prod namespaces from our existing preview environment

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,78 @@
...
+    * also adds `--preview` to kv commands in order to interact with them instead of production namespaces.
+
+    If you define a KV Namespace in your `wrangler.toml` but don't specify a `preview_id`, and then try to run `wrangler dev`, you'll see the following:
+    ```console
+    $ wrangler preview
+    💁  JavaScript project found. Skipping unnecessary build!
+    Error: In order to preview a worker with KV namespaces, you must designate a preview_id for each KV namespace you'd like to preview.
+    ```
+
+    More details can be found in the documentation.
```
Comment (EverlastingBugstopper) [inline CHANGELOG.md:24]: i still need to write the docs for this - aaannndd we should link to them

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,78 @@
...
+
+  - **`wrangler subdomain` allows you to rename your existing workers.dev subdomain - [nprogers], [issue/1279] [pull/1353]**
+
+    Workers recently introduced the ability to rename your workers.dev subdomain in the Workers dashboard. This change allows you to do the same thing from `wrangler`.
+
+    [nprogers]: https://github.com/nprogers
+    [pull/1353]: https://github.com/cloudflare/wrangler/pull/1353
+    [issue/1279]: https://github.com/cloudflare/wrangler/issues/1279
+
+  - **Only update wasmpack and cargo-generate when there's a new version - [jspspike], [issue/1240] [pull/1358]**
```
Comment (EverlastingBugstopper) [inline CHANGELOG.md:39]:
```suggestion
  - **Only update wasm-pack and cargo-generate when there's a new version - [jspspike], [issue/1240] [pull/1358]**
```

this one should probably go in maintenance section, and it should probably look like

```markdown
**Pin wasm-pack and cargo-generate to specific versions instead of always downloading the latest version - [jspspike], [issue/1240] [pull/1358]**

Previously, every time `wasm-pack` or `cargo-generate` was needed, Wrangler would download and install the latest version. Now, we pin those dependencies to a specific version in every release of Wrangler. If you have either of these tools installed locally, Wrangler will use them if you have a more recent version.
```

**Comment by ispivey** [comment]
Yeah, I hadn't made a PR yet because I have a lot left to do 😬  Thanks for the early feedback!

**Comment by EverlastingBugstopper** [comment]
ooops

File: CHANGELOG.md (hunk 3 above)
Comment (ispivey) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 2 above)
Comment (ispivey) [reply CHANGELOG.md:22]: Haha, yep, that's mostly why I left this to remind both of us!

File: CHANGELOG.md (hunk 1 above)
Comment (ispivey) [reply CHANGELOG.md]: Updated, thanks!

**Comment by ispivey** [comment]
@cloudflare/wrangler I think this is ready for re-review. I'm gonna poke at the CI bug a little while I'm here, and being annoyed at it.

**Comment by ispivey** [comment]
oh shoot I need to pick an emoji

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,139 @@
...
+
+  - **Pin wasm-pack and cargo-generate to specific versions instead of always downloading the latest version - [jspspike], [issue/1240] [pull/1358]**
+
+    Previously, every time `wasm-pack` or `cargo-generate` was needed, Wrangler would download and install the latest version. Now, we pin those dependencies to a specific version in every release of Wrangler. If you have either of these tools installed locally, Wrangler will use them if you have a more recent version.
+
+    [jspspike]: https://github.com/jspspike
+    [pull/1358]: https://github.com/cloudflare/wrangler/pull/1358
+    [issue/1240]: https://github.com/cloudflare/wrangler/issues/1240
+
+  - **Support batched KV PUT and DELETE operations to improve Workers Site publish performance - [jspspike], [issue/1191] [pull/1339]**
```
Comment (EverlastingBugstopper) [inline CHANGELOG.md:100]: mm - doesn't really improve performance. It just lets folks w/more than 10k static files use kv:bulk commands (and i think sites as well)

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,139 @@
...
+- ### Maintenance
+
+  - **Add Code of Conduct - [EverlastingBugstopper], [pull/1346]**
+
+    We actually didn't have one before, so now we do!
+
+    [EverlastingBugstopper]: https://github.com/EverlastingBugstopper
+    [pull/1346]: https://github.com/cloudflare/wrangler/pull/1346
+
+  - **Deprecate undocumented KV `bucket` attribute - [ashleymichal], [issue/1136] [pull/1355]**
```
Comment (EverlastingBugstopper) [inline CHANGELOG.md:117]: i wonder if we should put this above the fold since technically it's a breaking change

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,139 @@
...
+
+    The `bucket` attribute was not originally intended to be released into the wild. We announced deprecation in March, and are now removing it as an available configuration property.
+
+    [ashleymichal]: https://github.com/ashleymichal
+    [pull/1355]: https://github.com/cloudflare/wrangler/pull/1355
+    [issue/1136]: https://github.com/cloudflare/wrangler/issues/1136
+
+  - **Refactor to separate KV commands, implementation of said commands, and site-specific logic - [ashleymichal], [pull/1332]**
```
Comment (EverlastingBugstopper) [inline CHANGELOG.md:125]: wonder if we can come up with a better tagline here, not the worst but it's a bit wordy

File: CHANGELOG.md (hunk 4 above)
Comment (ispivey) [reply CHANGELOG.md:100]: Ooh, gotcha. Will update.

File: CHANGELOG.md (hunk 6 above)
Comment (ashleymichal) [reply CHANGELOG.md:125]: refactor allthethings

**Comment by ispivey** [comment]
I just rebased and force-pushed so I could run the test suite without nightly builds.

File: CHANGELOG.md (hunk 2 above)
Comment (EverlastingBugstopper) [reply CHANGELOG.md:22]: https://github.com/cloudflare/workers-docs/pull/904
